### PR TITLE
Document keyword and default arguments

### PR DESCRIPTION
**Pull request by TheZoq2**
I think keyword and default arugments may be new to some people so let's document those. I also took the liberty of adding some color to function definitions in the docs to make things more readable

image

And here is what the new section looks like

image

### CODE REVIEW

File: docs/ribs.docs [hunk 1]
```diff
@@ -1,10 +1,46 @@
...
 Ribs is a super small and easy to use collection of functions
 to help with some things pygame don't help with.
 
 Also note that not everything is documented, if it's not
 documented you probably don't have to care about it.
 
+# Keyword and default arguments
+
+<p>
+Some of the functions in ribs use a python feature you may not have come across: default arguments. These allow you to only specify a few of the arguments to a function, and leave the rest. For example, the definition of <code>draw_transformed</code> looks like this:
```
Comment (sornas) [inline docs/ribs.docs]:
```suggestion
Some of the functions in ribs use a Python feature you may not have come
across: default arguments. These allow you to only specify a few of the
arguments to a function, and leave the rest. For example, the definition of
<code>draw_transformed</code> looks like this:
```

File: docs/ribs.docs [hunk 2]
```diff
@@ -1,10 +1,46 @@
 # Notes about RIBS
+<p>
+Ribs is super small and easy to use collection of functions to help make pygame
+easier to work with. This includes things like drawing images in a single function,
+handling user input, and a few other things.
```
Comment (sornas) [inline docs/ribs.docs]: This text (at least parts of it) already exists below this addition.

File: docs/ribs.docs [hunk 3]
```diff
@@ -1,10 +1,46 @@
...
+</p>
+<p>
+If you want do do something in your game but can't find a
+suitable function, you may want use pygame directly. All documentation for the pygame
+can be found at <a href="https://www.pygame.org/docs/">https://www.pygame.org/docs/</a>
+</p>
+<p>
+Also note that not everything is documented, if it's not documented you
+probably don't have to care about it.
```
Comment (sornas) [inline docs/ribs.docs:11]: This text already exists below this addition.

File: docs/ribs.docs [hunk 4]
```diff
@@ -1,10 +1,46 @@
 # Notes about RIBS
+<p>
+Ribs is super small and easy to use collection of functions to help make pygame
+easier to work with. This includes things like drawing images in a single function,
+handling user input, and a few other things.
+</p>
+<p>
+If you want do do something in your game but can't find a
+suitable function, you may want use pygame directly. All documentation for the pygame
+can be found at <a href="https://www.pygame.org/docs/">https://www.pygame.org/docs/</a>
```
Comment (sornas) [inline docs/ribs.docs]:
```suggestion
If you want do do something in your game but can't find a
suitable function, you may want use pygame directly. All documentation for pygame
can be found at <a href="https://www.pygame.org/docs/">https://www.pygame.org/docs/</a>
```

File: docs/doc-generator.py [hunk 5]
```diff
@@ -16,6 +16,56 @@ def gen_id(string):
...
+                # If this is a new argument
+                arg_regex = re.search(r"([A-z]+)(=?)(.*)", part)
+                if arg_regex is not None:
+                    arg_name = arg_regex.group(1)
+                    if arg_regex.group(2) == '':
+                        # If this is not a keyword arg, add it to the list:
+                        args.append((arg_name, None))
+                    else:
+                        # This is a keyword arg, add it, along with a list of 
+                        args.append((arg_name, [arg_regex.group(3)]))
```
Comment (sornas) [inline docs/doc-generator.py]: Trailing whitespace or cut-off comment?

File: docs/doc-generator.py [hunk 6]
```diff
@@ -16,6 +16,56 @@ def gen_id(string):
...
+            result = f"<span class='arg'>{name}</span>"
+            if default:
+                result += "<span class='eq'>=</span>"
+                result += "<span class='default_arg'>"
+                result += ", ".join(default)
+                result += "</span>"
+            return result
+
+        arg_list = []
+
```
Comment (sornas) [inline docs/doc-generator.py]: Unused variable.

**Review by FredTheDino — APPROVED** [review]
LGTM, cool functionality!

File: docs/doc-generator.py [hunk 7]
```diff
@@ -16,6 +16,56 @@ def gen_id(string):
...
+        args = []
+        if arg_string:
+            # Try to build a list of arguments
+            comma_separated = arg_string.split(',')
+            # Spliting on , has side effects when the default value has , in it. We'll try
+            # to rectify that by re-joining string which don't look like identifiers
+            for part in comma_separated:
+                # If this is a new argument
+                arg_regex = re.search(r"([A-z]+)(=?)(.*)", part)
```
Comment (sornas) [inline docs/doc-generator.py]: If I understand this correctly, the regex looks for parameters and potential default values. If so, you want something like this:
```suggestion
                # If this is a new argument
                arg_regex = re.search(r"([A-z_][0-9A-z_]*)(=?)(.*)", part)
```

File: docs/doc-generator.py [hunk 8]
```diff
@@ -16,6 +16,56 @@ def gen_id(string):
...
+        result  = f"<span class='fn_name'>{fn_name}</span>"
+        result += f"<span class='paren'>( </span>"
+        result += "<span class='comma'>, </span>".join([format_arg(x) for x in args])
+        result += f"<span class='paren'> )</span>"
+        return result
+    return name
+
+
+
+
```
Comment (sornas) [inline docs/doc-generator.py]: Bit too many empty lines here.

**Comment by sornas** [comment]
Explaining named and default arguments is a great idea.

**Comment by FredTheDino** [comment]
I think I fixed the issues you found in @TheZoq2 s code, let me know if you find anything more

**Review by sornas — APPROVED** [review]
Looks good :+1:

**Pull request merged**